source/codegen/metadata/nixnetsocket/custom_proto.mako

```diff
@@ -16,3 +16,27 @@ message SockAddr {
     SockAddrIPv6 ipv6 = 2;
   }
 }
+
+enum OptName {
```

Comment:
The majority of these came from the "Socket options" #define section in the header. Looking at SocketTests.cpp, however, showed a few more being used like TCP_NODELAY and IP_ and IPv6_ options. I only included the ones I saw in the example tests but we need to iron out this list and determine if it is exhaustive or not.

If it ends up not being exhaustive, I can keep it but make the SockOptData `opt_name` field  be a oneof to either input a known option or pass in a raw value for the option. Either way should work fine with the conversion for SetSockOpt since the client's setting the data type (bool, int, or string).

If it's not exhaustive though, for GetSockOpt I think I'll need an extra param indicating the type (since it can't be derived from opt_name) or seperate out into different methods for each type.